### PR TITLE
Hotfix v1.4.2 to resolve bug with ncells in feature dataframe #246

### LINKED ISSUE #246: `ncells` gets same value for all features in one timestep
Thanks to @laurapaccini who posted in #245  that `ncells` in the feature output from the segmentation do not always correspond to the number of grid cells associated with the feature in the segmentation mask. 

Eventually, I found the problem in the segmentation code, where we iterate over the different features in one time step, but assign only the value of `ncells` of the last feature to all features : 

https://github.com/tobac-project/tobac/blob/09b176b2037fe75ee79a2096d633e790579f02ef/tobac/segmentation.py#L267-L275

I double-checked this behavior in our example notebooks, which all show only one unique value of `ncells` per timestep, although there are multiple features detected in each timestep. 

This bug will be relatively easy to fix but should be done asap and preferably with a test making sure that this does not happen again. 

Since we are about to release `v1.5` maybe it suffices to include this in `v1.5`?  But I leave it to @w-k-jones and  @freemansw1 to judge if another hotfix is needed.

### PR DESCRIPTION
**Pull request by JuliaKukulies**
Resolves #246 

- fixed bug in `segmentation.py` that assigned `ncells` of the last feature in each time step to all features in that time step
- added a test to make sure that `ncells` is correctly assigned to different features
- started to prepare this as a hotfix -> version bump and changelog

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [x] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [ ] If you have introduced new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [ ] If you have introduced new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

File: tobac/segmentation.py
Comment (w-k-jones) [inline tobac/segmentation.py:273]: Could this be assigned directly rather than assigned a temporary variable?

File: tobac/tests/test_segmentation.py
Comment (w-k-jones) [inline tobac/tests/test_segmentation.py:3]: Could we change these to import the full names for clarity? e.g. `from tobac import segmentation, feature_detection`

**Review by w-k-jones — COMMENTED** [review]
Looks good. Comments are only minor changes that make no functional difference so can be left if you think that they are unnecessary

File: tobac/tests/test_segmentation.py
Comment (JuliaKukulies) [reply tobac/tests/test_segmentation.py]: Agreed, I changed this

File: tobac/segmentation.py
Comment (JuliaKukulies) [reply tobac/segmentation.py]: Done!

**Review by w-k-jones — APPROVED** [review]
Looks good to me, happy for this to be merged after second review

**Comment by w-k-jones** [comment]
Also happy to merge with a single review if you don't have time to review this @freemansw1

**Review by freemansw1 — APPROVED** [review]
Thanks for doing this @JuliaKukulies; sorry for my delay in reviewing, it's been a busy week.

**Comment by freemansw1** [comment]
@JuliaKukulies @w-k-jones I'm happy for either of you to do the release. @JuliaKukulies it might be good for you to do one, if you're up for it!

**Comment by JuliaKukulies** [comment]
> @JuliaKukulies @w-k-jones I'm happy for either of you to do the release. @JuliaKukulies it might be good for you to do one, if you're up for it!

Thanks @freemansw1 and no worries! I am happy to do the release - only question is if the bug fixes in #250 or even #244 should be included or if we do the hotfix release just for this PR ?

**Comment by freemansw1** [comment]
I'll leave that to @w-k-jones. I think at least #244 should be included, but I'll defer to his judgment.

**Comment by w-k-jones** [comment]
Although I'd like to include #244 and #250 in 1.4.2, both changes were made on the v1.5 branch so I'm not sure how straightforward they would be to transfer

**Comment by w-k-jones** [comment]
On second inspection, I think that both will be straightforward to transfer, so I'll make a new PR with the changes against the hotfix branch

**Comment by JuliaKukulies** [comment]
All right, merging this now. Note that I have already updated the changelog here, which you can have a final check on before merging `hotfix` into `main`

**Pull request merged**
